app/uk/gov/hmrc/gform/gform/FormController.scala

```diff
@@ -81,11 +81,15 @@ class FormController(
 
   def dashboard(formTemplateId: FormTemplateId, lang: Option[String]) = auth.async(formTemplateId, lang) {
     implicit request => cache =>
-      (cache.formTemplate.draftRetrievalMethod, cache.retrievals.affinityGroup) match {
-        case (Some(FormAccessCodeForAgents), Some(AffinityGroup.Agent)) =>
-          Future.successful(Ok(access_code_start(cache.formTemplate, AgentAccessCode.form, lang, frontendAppConfig)))
-        case _ => Future.successful(Redirect(routes.FormController.newForm(formTemplateId, lang)))
-      }
+      val formId = FormId(cache.retrievals.userDetails, formTemplateId, None)
+      for {
+        maybeForm <- gformConnector.maybeForm(formId)
+      } yield
+        (cache.formTemplate.draftRetrievalMethod, cache.retrievals.affinityGroup, maybeForm.isDefined) match {
```

Comment:
Actually, using isDefined here is not a good pattern, sorry.  Better to match mayBeForm with Some(_)